lib/importer/sequence_ontology_mirror.rb

```diff
@@ -33,8 +35,12 @@ def populate_ontology_entry
     end
 
     def valid_entry?(entry)
-      ['id', 'name'].inject(true) do |val, term|
-        entry[term].present? && val
+      if entry['is_obsolete'].present?
+        obsolete_terms.append(entry)
+      else
+        ['id', 'name'].inject(true) do |val, term|
```

Comment:
Dunno if it reads as any clearer to you, but this could also be `['id', 'name'].all? { |term| entry[term].present? }`what you've got will work just fine as well though. (It looks like you didn't actually make that change in the first place, its just formatting so don't feel like you have to change it)